packages/strapi-admin/admin/src/containers/Users/EditPage/index.js

```diff
@@ -1,19 +1,191 @@
-import React from 'react';
+import React, { useEffect, useReducer } from 'react';
 import { useRouteMatch } from 'react-router-dom';
-import { useGlobalContext } from 'strapi-helper-plugin';
+import { useIntl } from 'react-intl';
+import { get, isEmpty } from 'lodash';
+import {
+  // request,
+  useGlobalContext,
+} from 'strapi-helper-plugin';
+import { Col } from 'reactstrap';
+import { Padded } from '@buffetjs/core';
+import BaselineAlignement from '../../../components/BaselineAlignement';
+import ContainerFluid from '../../../components/ContainerFluid';
+import FormBloc from '../../../components/FormBloc';
+import SizedInput from '../../../components/SizedInput';
+import Header from '../../../components/Users/Header';
+import MagicLink from '../../../components/Users/MagicLink';
+import SelectRoles from '../../../components/Users/SelectRoles';
+import { editValidation } from '../../../validations/users';
+import checkFormValidity from '../../../utils/checkFormValidity';
+import form from './utils/form';
+import fakeData from './utils/tempData';
+import { initialState, reducer } from './reducer';
+import init from './init';
 
 const EditPage = () => {
   const { settingsBaseURL } = useGlobalContext();
+  const [
+    { formErrors, isLoading, initialData, modifiedData, showHeaderLoader },
+    dispatch,
+  ] = useReducer(reducer, initialState, init);
+  const { formatMessage } = useIntl();
   const {
     params: { id },
   } = useRouteMatch(`${settingsBaseURL}/users/:id`);
+  const headerLabelId = isLoading
+    ? 'app.containers.Users.EditPage.header.label-loading'
+    : 'app.containers.Users.EditPage.header.label';
+  const headerLabelName = initialData.username
+    ? initialData.username
+    : `${initialData.firstname} ${initialData.lastname}`;
+  const headerLabel = formatMessage({ id: headerLabelId }, { name: headerLabelName });
+
+  useEffect(() => {
```

Comment:
This component is very similar to the `<ProfilePage />` I recommend that we create a hook `useUserForm` that handles the fetch & submit logic, as well as the form and validations ones.